### PR TITLE
RavenDB-15979: tighten subscription test

### PR DESCRIPTION
**Pull request by garayx**

### CODE REVIEW

File: src/Raven.Client/Documents/Subscriptions/SubscriptionWorker.cs [hunk 1]
```diff
@@ -198,8 +198,10 @@ await requestExecutor.ExecuteAsync(_redirectNode, null, context, command, should
                     {
                         await requestExecutor.ExecuteAsync(command, context, sessionInfo: null, token: token).ConfigureAwait(false);
                         tcpInfo = command.Result;
-
-                        _redirectNode = requestExecutor.Topology.Nodes.Where(x => tcpInfo.Urls.Contains(x.Url)).FirstOrDefault();
+                        if (tcpInfo.NodeTag != null)
+                        {
+                            _redirectNode = requestExecutor.Topology.Nodes.FirstOrDefault(x => x.ClusterTag == tcpInfo.NodeTag);
+                        }
```
Comment (ayende) [inline src/Raven.Client/Documents/Subscriptions/SubscriptionWorker.cs:204]: Can you explain this change?

Comment (ppekrol) [reply src/Raven.Client/Documents/Subscriptions/SubscriptionWorker.cs:204]: We were comparing oranges to apples.
`Node.Url` is e.g. `localhost:8080` and `TcpInfo.Urls` are `localhost:12345;localhost:12346`

Comment (ayende) [reply src/Raven.Client/Documents/Subscriptions/SubscriptionWorker.cs:204]: Won't that mean that we are now breaking backward compact with servers that won't send this?

Comment (ppekrol) [reply src/Raven.Client/Documents/Subscriptions/SubscriptionWorker.cs:204]: It was always returning `null` anyway, so it should not be a problem?

**Review by ayende — APPROVED** [review]

**Pull request merged**